lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -88,6 +128,57 @@ class DecomposeAtenSoftmaxIntOp : public OpRewritePattern<AtenSoftmaxIntOp> {
 };
 } // namespace
 
+// Aten_SoftmaxBackwardDataOp(gradOutput, output, dim) =>
+//    newGrad = gradOutput * output
+//    result = newGrad - output * sum(newGrad, dim))
+//
+// Refer to
+// https://github.com/pytorch/pytorch/blob/15fecc4c830a3907fde4b44c9962dc4144da50a4/torch/csrc/jit/codegen/cuda/ops/normalization.cpp#L31
+namespace {
+class DecomposeAten_SoftmaxBackwardDataOp
+    : public OpRewritePattern<Aten_SoftmaxBackwardDataOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten_SoftmaxBackwardDataOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    MLIRContext *context = op.getContext();
+    Value gradOutput = op.grad_output();
+    Value output = op.output();
+    Value dim = op.dim();
+
+    BaseTensorType tensorType = gradOutput.getType().cast<BaseTensorType>();
+    if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
+      return rewriter.notifyMatchFailure(op, "Only support floating type");
+
+    // newGrad = gradOutput * output
```

Comment:
Do we need these comments? The function is already described above.

Reply:
removed as there is just one operation and easy to read.